coresimd/ppvt/api/hash.rs

```diff
@@ -0,0 +1,50 @@
+//! Implements `Hash`.
+
+macro_rules! impl_hash {
+    ($id:ident, $elem_ty:ident) => {
+        impl hash::Hash for $id {
+            #[inline(always)]
+            fn hash<H: hash::Hasher>(&self, state: &mut H) {
+                union A {
+                    data: [$elem_ty; $id::lanes()],
+                    vec: $id
+                }
+                unsafe {
+                    let mut bytes: A = mem::uninitialized();
+                    self.store_aligned_unchecked(&mut bytes.data);
```

Comment:
Could this function be simplified to:

```rust
A { vec: *self }.data.hash(state)
```

Reply:
I'll give it a try.

I should fill an issue to discuss the semantics of this, it basically just has the vector as a slice of bytes. This is a bit differently of how `Hash` works for arrays, where first the length is hashes, and then each element is hashed.

Reply:
Done.